src/pages/index.js

```diff
@@ -1,93 +1,225 @@
+        },
+      },
+      selectors.cardList
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+// Validation
 
-// VALIDATION //
 const formValidators = {};
 
 const enableValidation = (config) => {
   const formList = Array.from(document.querySelectorAll(config.formSelector));
   formList.forEach((formElement) => {
     const validator = new FormValidator(config, formElement);
     const formName = formElement.getAttribute("name");
-
-    formValidators[formName] = validator;
-    validator.enableValidation();
+    if (formName) {
+      formValidators[formName] = validator;
+      validator.enableValidation();
+    }
   });
 };
 
-//Event Listeners
-profileEditButton.addEventListener("click", () => {
-  fillProfileForm();
-  formValidators[selectors.profileForm].resetFormValidation();
-  userInfoPopup.open();
-});
+// Create Instances
 
-addNewCardButton.addEventListener("click", () => {
-  newCardPopup.open();
+const cardPreviewPopup = new PopupWithImage(selectors.previewPopup);
+
+const userInfo = new UserInfo(
+  selectors.profileTitle,
+  selectors.profileParagraph,
+  selectors.profileImage
+);
+
+const deleteCardPopup = new PopupWithConfirm(selectors.deleteCardPopup);
+
+const addCardPopup = new PopupWithForm(selectors.addFormPopup, (cardData) => {
+  addCardPopup.setLoading(true);
+  api
+    .createNewCard(cardData)
+    .then((res) => {
+      const card = renderCard(res);
+      cardSection.addItem(card);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardPopup.setLoading(false);
+    });
 });
 
+const ProfileEditPopup = new PopupWithForm(
+  selectors.editFormPopup,
+  (userData) => {
+    ProfileEditPopup.setLoading(true);
+    api
+      .updateUserInfo(userData)
+      .then(() => {
+        userInfo.setUserInfo(userData.name, userData.about);
+        ProfileEditPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        ProfileEditPopup.setLoading(false);
+      });
+  }
+);
+
+const updateAvatarPopup = new PopupWithForm(
+  selectors.changeImagePopup,
+  (data) => {
+    updateAvatarPopup.setLoading(true);
+    api
+      .updateAvatar(data.link)
+      .then((user) => {
+        userInfo.setAvatar(user.avatar);
+        updateAvatarPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        updateAvatarPopup.setLoading(false);
+      });
+  }
+);
+
 // Functions
 
 function fillProfileForm() {
-  const { name, description } = userInfo.getUserInfo();
+  const { name, about } = userInfo.getUserInfo();
   document.querySelector(selectors.inputName).value = name;
-  document.querySelector(selectors.inputDescription).value = description;
+  document.querySelector(selectors.inputDescription).value = about;
+}
+
+function handleLikeClick(card) {
+  if (card._isLiked) {
```

Comment:
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.